beacon-chain/rpc/eth/validator/handlers.go

```diff
@@ -556,40 +551,38 @@ func (s *Server) PrepareBeaconProposer(w http.ResponseWriter, r *http.Request) {
 		httputil.HandleError(w, "Could not decode request body: "+err.Error(), http.StatusBadRequest)
 		return
 	}
-	var feeRecipients []common.Address
 	var validatorIndices []primitives.ValidatorIndex
 	// filter for found fee recipients
 	for _, r := range jsonFeeRecipients {
 		validatorIndex, valid := shared.ValidateUint(w, "validator_index", r.ValidatorIndex)
 		if !valid {
-			return
+			log.WithField("validatorIndex", validatorIndex).Warn("fee recipient is invalid")
+			continue
 		}
 		feeRecipientBytes, valid := shared.ValidateHex(w, "fee_recipient", r.FeeRecipient, fieldparams.FeeRecipientLength)
 		if !valid {
-			return
+			log.WithField("validatorIndex", validatorIndex).Warn("fee recipient is invalid")
+			continue
 		}
-		f, err := s.BeaconDB.FeeRecipientByValidatorID(ctx, primitives.ValidatorIndex(validatorIndex))
-		switch {
-		case errors.Is(err, kv.ErrNotFoundFeeRecipient):
-			feeRecipients = append(feeRecipients, common.BytesToAddress(bytesutil.SafeCopyBytes(feeRecipientBytes)))
-			validatorIndices = append(validatorIndices, primitives.ValidatorIndex(validatorIndex))
-		case err != nil:
-			httputil.HandleError(w, fmt.Sprintf("Could not get fee recipient by validator index: %v", err), http.StatusInternalServerError)
-			return
-		default:
-			if common.BytesToAddress(feeRecipientBytes) != f {
-				feeRecipients = append(feeRecipients, common.BytesToAddress(bytesutil.SafeCopyBytes(feeRecipientBytes)))
-				validatorIndices = append(validatorIndices, primitives.ValidatorIndex(validatorIndex))
+		// Use default address if the burn address is return
+		feeRecipient := primitives.ExecutionAddress(feeRecipientBytes)
+		if feeRecipient == primitives.ExecutionAddress([20]byte{}) {
+			feeRecipient = primitives.ExecutionAddress(params.BeaconConfig().DefaultFeeRecipient)
+			if feeRecipient == primitives.ExecutionAddress([20]byte{}) {
+				log.WithField("validatorIndex", validatorIndex).Warn("fee recipient is the burn address")
 			}
 		}
+		val := cache.TrackedValidator{
+			Active:       true, // TODO: either check or add the field in the request
+			Index:        primitives.ValidatorIndex(validatorIndex),
+			FeeRecipient: feeRecipient,
+		}
+		s.TrackedValidatorsCache.Set(val)
+		validatorIndices = append(validatorIndices, primitives.ValidatorIndex(validatorIndex))
```

Comment:
I think duplicate indexes need to be filtered somehow. is this being handled?

Reply:
I think the caller handles this but if it's through rest maybe we want to throw some logs if the same index is used with different fee recipients or  duplicates of the same one are sent.